core/domain/stats_jobs_one_off.py

```diff
@@ -18,14 +18,466 @@
+                            {
+                                state_name: {
+                                    'total_answers_count_v1',
+                                    'useful_feedback_count_v1',
+                                    'total_hit_count_v1',
+                                    'first_hit_count_v1',
+                                    'num_completions_v1'
+                                }
+                            }
+                    }
+        """
+        reduced_state_stats_mapping = {
+            state_name: {
+                'total_answers_count_v1': item.state_stats_mapping[state_name][
+                    'total_answers_count_v1'],
+                'useful_feedback_count_v1': item.state_stats_mapping[
+                    state_name]['useful_feedback_count_v1'],
+                'total_hit_count_v1': item.state_stats_mapping[state_name][
+                    'total_hit_count_v1'],
+                'first_hit_count_v1': item.state_stats_mapping[state_name][
+                    'first_hit_count_v1'],
+                'num_completions_v1': item.state_stats_mapping[state_name][
+                    'num_completions_v1']
+            } for state_name in item.state_stats_mapping
+        }
+
+        yield (item.exp_id, {
+            'exp_version': item.exp_version,
+            'num_starts_v1': item.num_starts_v1,
+            'num_completions_v1': item.num_completions_v1,
+            'num_actual_starts_v1': item.num_actual_starts_v1,
+            'state_stats_mapping': reduced_state_stats_mapping
+        })
+
+    @staticmethod
+    def reduce(exp_id, stringified_values):
+        for exp_stats in stringified_values:
+            exp_stats = ast.literal_eval(exp_stats)
+            exp_version = exp_stats['exp_version']
+
+            num_starts_v1 = exp_stats['num_starts_v1']
+            num_completions_v1 = exp_stats['num_completions_v1'],
+            num_actual_starts_v1 = exp_stats['num_actual_starts_v1']
+
+            StatisticsAuditV1.require_non_negative(
+                exp_id, exp_version, 'num_starts_v1', num_starts_v1)
+            StatisticsAuditV1.require_non_negative(
+                exp_id, exp_version, 'num_completions_v1', num_completions_v1)
+            StatisticsAuditV1.require_non_negative(
+                exp_id, exp_version, 'num_actual_starts_v1',
+                num_actual_starts_v1)
+
+            # Number of starts must never be less than the number of
+            # completions.
+            if num_starts_v1 < num_completions_v1:
+                yield ('Completions > starts: exp_id:%s version:%s %s > %s' % (
+                    exp_id, exp_version, num_completions_v1, num_starts_v1),)
+
+            # Number of actual starts must never be less than the number of
+            # completions.
+            if num_actual_starts_v1 < num_completions_v1:
+                yield ('Completions > actual starts: exp_id:%s version:%s %s > %s' % ( # pylint: disable=line-too-long
```

Comment:
Refrain from pragmas, here and below. Instead try something like:

    yield (
        'Completions > ...'
        '...' % (
            exp_id, exp_version, ...),)

Reply:
Done.